cloudify_agent/installer/config/configuration.py

```diff
@@ -165,6 +171,17 @@ def _cfy_agent_attributes_no_defaults(cloudify_agent):
         # by default, the manager ip will be set by an environment variable
         cloudify_agent['manager_ip'] = get_manager_ip()
 
+    security_enabled = ctx.security_ctx.security_enabled
```

Comment:
1) validate if security_enabled and not (username and password)
2) add `security_enabled` + certificate related